icu4c/source/i18n/unitconverter.cpp

```diff
@@ -15,6 +18,126 @@
 U_NAMESPACE_BEGIN
 
 namespace {
+
+/* Internal Structure */
+
+enum Constants {
+    CONSTANT_FT2M,    // ft2m stands for foot to meter.
+    CONSTANT_PI,      // PI
+    CONSTANT_GRAVITY, // Gravity
+    CONSTANT_G,
+    CONSTANT_GAL_IMP2M3, // Gallon imp to m3
+    CONSTANT_LB2KG,      // Pound to Kilogram
+
+    // Must be the last element.
+    CONSTANTS_COUNT
+};
+
+typedef enum SigNum {
+    NEGATIVE = -1,
+    POSITIVE = 1,
+} SigNum;
+
+/* Represents a conversion factor */
+struct Factor {
+    double factorNum = 1;
+    double factorDen = 1;
+    double offset = 0;
+    bool reciprocal = false;
+    int32_t constants[CONSTANTS_COUNT] = {};
+
+    void multiplyBy(const Factor &rhs) {
+        factorNum *= rhs.factorNum;
+        factorDen *= rhs.factorDen;
+        for (int i = 0; i < CONSTANTS_COUNT; i++)
+            constants[i] += rhs.constants[i];
+
+        offset += rhs.offset;
```

Comment:
OK.  Please add a comment here documenting that we're not sure what is the best way to handle offset in multiplyBy/divideBy, so that someone coming here in the future isn't afraid to change the logic to something better.